strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/main/java/org/carlspring/strongbox/repository/NugetRepositoryFeatures.java

```diff
@@ -115,4 +147,70 @@ protected Configuration getConfiguration()
         return configurationManager.getConfiguration();
     }
 
+    @Component
+    public class RepositorySearchEventListener implements CommonEventListener<RemoteRepositorySearchEvent>
+    {
+
+        @Override
+        public void handle(RemoteRepositorySearchEvent event)
+        {
+            RepositorySearchRequest repositorySearchRequest = event.getEventData();
+            Map<String, String> coordinates = repositorySearchRequest.getCoordinates();
+            String packageId = coordinates.get(NugetArtifactCoordinates.ID);
+            String version = coordinates.get(NugetArtifactCoordinates.VERSION);
+
+            Expression filter = repositorySearchRequest.isStrict() ? createPackageEq(packageId, null) : null;
+            filter = createVersionEq(version, filter);
+            String searchTerm = !repositorySearchRequest.isStrict() ? packageId : null;
+
+            try
+            {
+                downloadRemoteFeed(repositorySearchRequest.getStorageId(), repositorySearchRequest.getRepositoryId(),
+                                   filter,
+                                   searchTerm, null, repositorySearchRequest.getSkip(),
+                                   repositorySearchRequest.getLimit());
+            }
+            catch (RepositoryInitializationException | ArtifactTransportException e)
+            {
+                logger.error(String.format("Failed to fetch Nuget remote feed [%s]",
+                                           repositorySearchRequest.getCoordinates()),
+                             e);
+            }
+        }
+
+    }
+
+    public static Expression createVersionEq(String version,
+                                             Expression filter)
+    {
+        if (version == null || version.trim().isEmpty())
```

Comment:
`StringUtils.isBlank`

Reply:
I don't like to use StringUtils for some reason. Hope that you guys will excuse these to me :)

Reply:
I don't mind :)

Reply:
thanks :)